### ISSUE TITLE
Odd behavior with "finished" titles 

### ISSUE THREAD

**Issue by vertigo235** [nosmokingbandit/watcher#84]
I noticed something odd since switching to nzbhydra, a day or so later, the status of "finished" will be replaced with what appears to be the IMDB ID, so that causes watcher to snatch the next result and download the movie again.

This may or may not be related to nzbhydra, but I would say it probably isn't. There have also been watcher updates since I moved to NZBHydra.

NZBHydra is good at removing all the duplicates if you use multiple indexers, because watcher doesn't seem to do that yet and it will keep downloading the same post over and over.

**Comment by vertigo235** [comment]
Here is a screenshot of what I'm talking about, the status of this title was "finished" but it changed to this odd status. 

![image](https://cloud.githubusercontent.com/assets/1238461/21591786/2d84b186-d0d5-11e6-8200-47c68c7f58b2.png)

**Comment by soundjester** [comment]
Which downloader/post-processing script are you using?

**Comment by vertigo235** [comment]
nzbget with the supplied script (although modified to include the /watcher/ url base. 

but this isn't related to that at all, everything works properly, the show is snatched, it goes to nzbget, then nzbget calls the post process script.  watcher updates status to "finished" the banner says finished etc.  All is good, but it looks like when watcher searches for an upgrade (since it continues to look for hte best match), that's when the weirdness happens.  The finished status gets replaced with the IMDB ID and then it goes to either found or not available.

**Comment by soundjester** [comment]
I was thinking it might be database weirdness, where a field gets transposed in a subsequent and later executed update call. Oh well.

**Comment by vertigo235** [comment]
yeah I was thinking the same, I was browsing the code and nothing jumped out at me

I have not had time to play with watcher lately so I have not had it happen again, I turned off nzbhydra for now to see if it happens with my 3 other indexers enabled though

**Comment by soundjester** [comment]
I use hydra exclusively (*more indexers than my body has room for* /powerthirst style) and haven't run into it. It will be interesting to see if it comes down to a specific indexer/search provider.

**Comment by vertigo235** [comment]
Interesting then if you are using Hydra I'm going to go back to that then. It's kind of necessary for this due to all the duplicate downloads without it IMO.

**Comment by soundjester** [comment]
My setup could probably not function without NZBHydra.  What OS/setup are you using?  I would like to try to duplicate, if possible.

**Comment by vertigo235** [comment]
Easy to duplicate I'm using the docker image.

**Comment by vertigo235** [comment]
On synology that is.

**Comment by soundjester** [comment]
Welp.  Can't duplicate that I'm afraid.  Maybe @WilliamJBrown can chime in.

**Comment by vertigo235** [comment]
So it did again 

[image](https://cloud.githubusercontent.com/assets/1238461/21719814/a4430f82-d3ed-11e6-876e-2be87d238939.png)

**Comment by vertigo235** [comment]
Here is the only thing in the logs, doesn't give too much indication on why it actually happened. 

INFO 2017-01-06 05:47:47,718 core.searcher.auto_search_and_grab: Jerry Maguire is Finished but Keep Searching is enabled. Checking if Finished date is less than 4 days ago.
INFO 2017-01-06 05:47:47,718 core.searcher.auto_search_and_grab: Jerry Maguire finished on 2017-01-06, searching again.
INFO 2017-01-06 05:47:47,719 core.newznab.search_all: SEARCHING: https://nzbhydra.blubeacon.com/api?apikey=APIKEY&t=movie&imdbid=0116695
INFO 2017-01-06 05:47:49,369 core.sqldb.get_search_results: Retreving Search Results for tt0116695.
INFO 2017-01-06 05:47:49,373 core.sqldb.get_movie_details: Retreving details for tt0116695.
INFO 2017-01-06 05:47:49,381 core.sqldb.get_marked_results: Retreving Marked Results for tt0116695.
INFO 2017-01-06 05:47:49,383 core.searcher.store_results: 10 results found for tt0116695. Storing results.
INFO 2017-01-06 05:47:50,012 core.sqldb.write_search_results: Writing batch into SEARCHRESULTS
INFO 2017-01-06 05:47:50,489 core.sqldb.get_distinct: Getting distinct values for tt0116695 in SEARCHRESULTS
INFO 2017-01-06 05:47:50,491 core.updatestatus.movie_status: Setting MOVIES tt0116695 status to Found.
INFO 2017-01-06 05:47:50,491 core.sqldb.update: Updating tt0116695 to Found in MOVIES.

**Comment by soundjester** [comment]
Up for some command-line shenanigans?

From the watcher directory:

```
1. sqlite3
2. attach "watcher.sqlite" as watcher;
3. .headers ON
4. .mode columns
5. select score, category, status, pubdate, imdbid, indexer, type from watcher.SEARCHRESULTS where imdbid='tt0116695';
```

See if the "status" column contains the IMDB id.

**Comment by vertigo235** [comment]
I can already answer that, yes it does.  I was checking out the DB the other day when I saw this.

**Comment by vertigo235** [comment]
I completely reinstalled the docker image from scratch with the latest. 

I am still experiencing the same issue on searches for upgrades, for some reason I think it might be this bit of code that is the issue, but I'm not sure, in the searcher.py module. 

    def search(self, imdbid, title):
        ''' Search indexers for releases
        :param imdbid: str imdb identification number (tt123456)
        :param title: str movie title and year (Movie Title 2016)

        Checks if guid matches entries in MARKEDRESULTS and
            sets status if found. Default status Available.

        Sends results to self.scoreresults(), then stores them in SEARCHRESULTS

        Returns Bool if movie is found.
        '''

        newznab_results = self.nn.search_all(imdbid)
        scored_results = self.score.score(newznab_results, imdbid, 'nzb')
        # TODO eventually add search for torrents

        # sets result status based off marked results table
        marked_results = self.sql.get_marked_results(imdbid)
        if marked_results:
            for result in scored_results:
                if result['guid'] in marked_results:
                    result['status'] = marked_results[result['guid']]

        if scored_results:
            if not self.store_results(scored_results, imdbid):
                return False

        if not self.update.movie_status(imdbid):
            logging.info('No acceptable results found for {}'.format(imdbid))
            return False

        return True

**Comment by nosmokingbandit** [comment]
This seems to be mostly a docker problem. I'm not sure why it transposes values like that, but I don't want to be involved in docker problems.

I'm going to close this, but feel free to keep working on it here among yourselves.

**Comment by vertigo235** [comment]
I'm not sure why it would have anything to do with Docker, just to make sure you are using the "continue searching for X days" feature yourself correct?  I had mine set for 0 days wait and then continue for 4 days.

**Comment by vertigo235** [comment]
I'll see if I can reproduce on my windows machine.

**Comment by nosmokingbandit** [comment]
I didn't read *all* of the comments, so I didn't notice that you were talking about a different issue altogether.

The upgrade quality search is the auto_search_and_grab() method of Searcher since it is strictly part of the automated process.

This is the important block

```

if status == 'Finished' and keepsearching == 'true':
                logging.info(u'{} is Finished but Keep Searching is enabled. Checking if Finished date is less than {} days ago.'.format(title, keepsearchingdays))
                finisheddateobj = datetime.datetime.strptime(finisheddate, '%Y-%m-%d').date()
                if finisheddateobj + keepsearchingdelta >= today:
                    logging.info('{} finished on {}, searching again.'.format(title, finisheddate))
                    self.search(imdbid, title)
                    continue
                else:
                    logging.info(u'{} finished on {} and is not within the search window.'.format(title, finisheddate))
                    continue
            continue

```

If the status field in the DB is not being stored correctly it won't continue searching/grabbing. Only if `status == 'Finished'`

Check to see what the actual value of the Status field is in the database.

**Comment by vertigo235** [comment]
OK so it looks like the movie table gets updated to finished status, but the MARKEDRESULTS table has the imdb ID as status after the download is completed. 

Waiting to see what happens after it searches for an upgrade.

**Comment by vertigo235** [comment]
OK I think I found the issue, in postprocessing.py 

like line 421
        if self.update.markedresults(data['guid'], data['imdbid'], 'Finished'):
            guid_result['update_MARKEDRESULTS'] = 'true'

I believe this calls updatestatus.py
   def markedresults(self, guid, status, imdbid=None):

And is changing the status on the MARKEDSTATUS table to be the imdb ID. 

At least I think, gotta put the computer away for a while.

ex

**Comment by WilliamJBrown** [comment]
Just want to add that I had an instance where this just happened to me. Not sure what caused it.

**Comment by vertigo235** [comment]
I'm certain the issue is above, there are several calls to update.markedresults where the imdb and status are in the wrong spot they are swapped

**Comment by vertigo235** [comment]
I sent a pull request, please forgive me if I did something wrong, I'm just learning github.

**Comment by vertigo235** [comment]
hey @nosmokingbandit , I saw you have made some updates but it looks like you didn't fix the inccrect calls to this method?  Did you see the above comments, or did you miss them because the issue is "closed" ?

**Comment by nosmokingbandit** [comment]
I had changed a few other things in postprocessing.py that caused conflicts, so I fixed it without pulling your request. I usually have a few things going at once so it is rare that a pull request wouldn't conflict with changes in my Testing branch.

That and the fact that the imdbid is a kwarg that we both managed to miss, so it wasn't doing anything being passed as a positional arg.

If you do a pull requests it is always wise to request it in the Testing branch, and make sure you aren't committing any more files than necessary. I woudln't want to pull your core/__init__.py since it points toward your fork, and your .settings folder (probably from whatever editor you use) wasn't ignored.

If you plan on using Git more I very strongly suggest [GitKraken](https://www.gitkraken.com/). Their website is kind of dumb, but it is the best Git manager I've used. Keeping track of commits it soooooo much easier with it.

Anyway, this should be all taken care of.
